crypto.go

```diff
@@ -109,18 +131,27 @@ func (s dhkemScheme) AuthEncap(rand io.Reader, pkR KEMPublicKey, skS KEMPrivateK
 		return nil, nil, err
 	}
 
-	zzER, err := s.group.DH(skE, pkR)
+	dhER, err := s.group.DH(skE, pkR)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	zzIR, err := s.group.DH(skS, pkR)
+	dhIR, err := s.group.DH(skS, pkR)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	zz := append(zzER, zzIR...)
-	return zz, s.group.Marshal(pkE), nil
+	dh := append(dhER, dhIR...)
+
+	enc := s.group.Marshal(pkE)
+	pkRm := s.group.Marshal(pkR)
+	pkSm := s.group.Marshal(skS.PublicKey())
+	kemContext := append(append(enc, pkRm...), pkSm...)
```

Comment:
Gotta love Go concatenation.  If you wanted to be a bit more frugal about allocations, you could alloc one buffer and then `copy` into it.  You could even give `Marshal` a slice argument to marshal into, à la `hash.Sum()`.

Reply:
Moved to `copy` calls!